include/HelicityFormalism.h

```diff
@@ -98,7 +98,11 @@ class HelicityFormalism : public SpinAmplitudeCache
     /// \param L orbital angular momentum
     /// \param two_S 2 * the total spin angular momentum
     virtual std::shared_ptr<SpinAmplitude> create(unsigned two_J, const SpinVector& two_j, unsigned l, unsigned two_s) const override
-    { return (two_J == 0) ? unit(two_J, two_j, l, two_s) : std::shared_ptr<SpinAmplitude>(new HelicitySpinAmplitude(two_J, two_j, l, two_s)); }
+    {
+        if (two_J == 0 and std::all_of(two_j.begin(), two_j.end(), [](int i){ return i == 0; }))
```

Comment:
For the helicity formalism, we have a constant spin amplitude whenever `J == 0` regardless of the daughter spins. Since `HelicitySpinAmplitude` already calculates the proper coefficient, we can just remove these lines and always return a `HelicitySpinAmplitude`.

We should modify the constructor of `HelicitySpinAmplitude` to have:
```cpp
addAmplitude(two_M, two_m, (initialTwoJ() == 0));
```
So that we don't create a `CachedValue` when the spin amplitude is constant.

`HelicitySpinAmplitude` should also override `SpinAmplitude::amplitude(...)` to just return back the coefficient if `J==0`, else to call `SpinAmplitude::amplitude(...)`; and override `SpinAmplitude::calculate(...)` to do nothing, if `J==0`, else to call `SpinAmplitude::calculate(...)`